homeassistant/components/recorder/migration.py

```diff
@@ -55,13 +55,11 @@ def get_schema_version(instance: Any) -> int:
         )
         current_version = getattr(res, "schema_version", None)
 
-        if current_version is None:
-            current_version = _inspect_schema_version(instance.engine, session)
-            _LOGGER.debug(
-                "No schema version found. Inspected version: %s", current_version
-            )
+    if current_version is None:
+        current_version = _inspect_schema_version(instance)
+        _LOGGER.debug("No schema version found. Inspected version: %s", current_version)
 
-        return current_version
+    return current_version
```

Comment:
If we call `_inspect_schema_version` within the scope of the session, the session doesn't end up calling `MutexPool._do_return_conn`; there's a lingering connection even after the recorder has shut down.

Not sure if this is an actual problem in the implementation or a problem with the tests only.